# 01 - Data Collection

This notebook loads and inspects the primary dataset for the hydride property prediction project.

## Data sources
- **ML-HydPARK** (Zenodo, v0.0.5): cleaned experimental thermodynamic data for metal hydrides (~770 entries)
- **ElementalH_Ef** and **ElementalH_Ef_MP**: elemental features from Materials Project, sourced from HEAhydrideML (Witman et al.)

## Goals
- Load and inspect each dataset
- Understand structure, columns, and data types
- Identify target variables and potential features
- Flag missing values and anomalies

In [1]:
import pandas as pd

df = pd.read_csv('../data/raw/ML-HYDPARK_v0.0.5.csv')
print(df.shape)
df.head()

(772, 13)


,Unnamed: 0,index,Material_Class,Composition_Formula,Hydrogen_Weight_Percent,Heat_of_Formation_kJperMolH2,Temperature_oC,Pressure_Atmospheres_Absolute,Entropy_of_Formation_JperMolH2perK,Equilibrium_Pressure_25C,LnEquilibrium_Pressure_25C,HtoM,Reference
0,0,0,A2B,Th2Al,0.8,130.0,500.0,0.001,110.711134,1.016736e-17,-39.127349,1.309571,NaN
1,1,1,A2B,Ti2Cu,2.2,130.0,500.0,0.120,150.515102,1.220083e-15,-34.339857,1.184850,NaN
2,2,2,A2B,Zr2Cu,1.3,144.0,600.0,0.003,116.621978,7.297612e-20,-44.064155,1.071443,NaN
3,3,3,A2B,Zr2Ni,1.3,183.0,604.0,0.003,160.332084,2.058957e-24,-54.539843,1.050307,NaN
4,4,4,A2B,Mg2Ni,3.6,64.5,299.0,3.200,122.403296,1.240158e-05,-11.297687,1.325126,NaN


In [2]:
df = pd.read_csv('../data/raw/ElementalH_Ef_MP.csv')
print(df.shape)
df.head()

(94, 4)


,Unnamed: 0,No,Species,Ef
0,0,1.0,NaN,NaN
1,1,2.0,NaN,NaN
2,2,3.0,Li,-0.482520
3,3,4.0,Be,-0.149158
4,4,5.0,B,-0.174701


In [3]:
df = pd.read_csv('../data/raw/ElementalH_Ef.csv')
print(df.shape)
df.head()

(94, 3)


,No,Species,Ef
0,1.0,NaN,NaN
1,2.0,NaN,NaN
2,3.0,Li,-0.41
3,4.0,Be,-0.04
4,5.0,B,-0.08


## Summary

| Dataset | Rows | Columns | Purpose |
|---|---|---|---|
| ML-HydPARK | 772 | 13 | Primary dataset: targets + thermodynamic features |
| ElementalH_Ef_MP | 94 | 4 | Elemental formation energies from Materials Project (DFT) |
| ElementalH_Ef | 94 | 3 | Elemental formation energies from literature |

**Next step:** EDA on ML-HydPARK: distribution of target variables, material class breakdown, missing values.